frontend/public/components/secrets/subforms.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-undef */
+import * as React from 'react';
+import { DropableFileInput } from '../utils';
+import HTML5Backend from 'react-dnd-html5-backend';
+import { DragDropContextProvider, DragDropContext } from 'react-dnd';
+
+export const SSHAuthSubform = DragDropContext(HTML5Backend)(class SSHAuthSubform extends React.Component<SSHAuthSubformProps, SSHAuthSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      'ssh-privatekey': this.props.stringData['ssh-privatekey'] || '',
+    };
+    this.changeData = this.changeData.bind(this);
+    this.onFileChange = this.onFileChange.bind(this);
+  }
+  changeData(event) {
+    this.setState({
+      'ssh-privatekey': event.target.value
+    }, () => this.props.onChange(this.state));
+  }
+  onFileChange(fileData) {
+    this.setState({
+      'ssh-privatekey': fileData
+    }, () => this.props.onChange(this.state));
+  }
+  render() {
+    return <DragDropContextProvider backend={HTML5Backend}>
```

Comment:
I didn't think you were supposed to use both `DragDropContext` and `DragDropContextProvider`

Reply:
we need both, at least based on the docs and examples

Reply:
Can you point me to the doc or example? I'm looking at

> **As an alternative to the DragDropContext**, you can use the DragDropContextProvider element to set up React DnD for your application. Similar to the DragDropContext, this may be injected with a backend via the backend prop, but it also can be injected with a window object.

http://react-dnd.github.io/react-dnd/docs-drag-drop-context-provider.html

I could be wrong :)

Reply:
> Can you point me to the doc or example?
This is the [example](https://github.com/react-dnd/react-dnd/blob/master/packages/documentation/examples/06%20Other/Native%20Files/Container.tsx) I was digging from.

Looking on your docs once more :rocket: